spec/requests/api_registrations_spec.rb

```diff
@@ -299,35 +299,34 @@
     end
 
     it 'accepts competitors from the waiting list' do
-      waiting_list = FactoryBot.create(:waiting_list, holder: competition)
-      waiting_list.add(registration1.user_id)
-      waiting_list.add(registration2.user_id)
-      waiting_list.add(registration3.user_id)
+      waitlisted1 = FactoryBot.create(:registration, competition: competition)
+      waitlisted2 = FactoryBot.create(:registration, competition: competition)
+      waitlisted3 = FactoryBot.create(:registration, competition: competition)
```

Comment:
These don't have the `:waiting_list` trait, but I feel like they should? (Based on the fact that in the previous state of things, you had added them to the waiting list manually)

Reply:
Great catch - I've added assertions to check their status before the test runs as well